packages/flutter/lib/src/widgets/drag_target.dart

```diff
@@ -418,26 +423,53 @@ class _DragAvatar<T> extends Drag {
     _entry.markNeedsBuild();
     HitTestResult result = new HitTestResult();
     WidgetFlutterBinding.instance.hitTest(result, globalPosition + feedbackOffset);
-    _DragTargetState<T> target = _getDragTarget(result.path);
-    if (target == _activeTarget)
+
+    Iterable<_DragTargetState<T>> targets = _getDragTargets(result.path);
+
+    bool listsMatch = false;
+    if (targets.length >= _lastTargets.length && _lastTargets.isNotEmpty) {
+      listsMatch = true;
+      Iterator<_DragTargetState<T>> it = targets.iterator;
+      for(int i = 0; i < _lastTargets.length; i++) {
+        it.moveNext();
+        if (it.current != _lastTargets[i]) {
+          listsMatch = false;
+          break;
+        }
+      }
+    }
+
+    // If everything's the same, bail early.
+    if (listsMatch) {
       return;
-    if (_activeTarget != null)
-      _activeTarget.didLeave(data);
-    _activeTarget = target;
-    _activeTargetWillAcceptDrop = _activeTarget != null && _activeTarget.didEnter(data);
+    }
+
+    // Leave old targets.
+    _lastTargets.forEach((_DragTargetState<T> target) => target.didLeave(data));
+    _lastTargets.clear();
+
+    // Enter new targets.
+    _DragTargetState<T> newTarget = targets.firstWhere((_DragTargetState<T> target) {
```

Comment:
this would probably be clearer as an olde tyme loop, but you can leave it like this if you prefer it. :-)